src/code/z_map_disp.c

```diff
@@ -1,109 +1,1813 @@
+    s32 mapId;
+    s32 mapCompactId;
+    s32 dungeonMapRoomIter;
+    s32 var_s2;
+    s32 roomIter;
+    MinimapEntry* entry;
+    s32 var_s3;
+
+    heapNext = heap;
+    if ((D_801BEBB8.minimapList == NULL) || (D_801BEC1C == 0)) {
+        return heapNext;
+    }
+    D_801F56B0.rooms = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106408.s")
+    // loop for number of rooms
+    for (roomIter = 0; roomIter < D_801BEC1C; roomIter++) {
+        var_s2 = false;
+        entry = &D_801BEBB8.minimapList->entry[roomIter];
+        if (entry->mapId == 0xFFFF) {
+            continue;
+        }
+        mapCompactId = MapData_GetMapCompactId(entry->mapId);
+        if (mapCompactId == -1) {
+            continue;
+        }
+        for (dungeonMapRoomIter = 0; dungeonMapRoomIter < D_801F56B0.rooms; dungeonMapRoomIter++) {
+            if (mapCompactId == D_801F56B0.unk4[dungeonMapRoomIter]) {
+                var_s2 = true;
+                break;
+            }
+        }
+        if (var_s2 == false) {
+            D_801F56B0.unk4[D_801F56B0.rooms] = mapCompactId;
+            D_801F56B0.rooms += 1;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106450.s")
+    D_801F56B0.unk84[0] = heap;
+    for (dungeonMapRoomIter = 0; dungeonMapRoomIter < D_801F56B0.rooms; dungeonMapRoomIter++) {
+        mapId = D_801F56B0.unk4[dungeonMapRoomIter];
+        MapDisp_GetMapITexture(D_801F56B0.unk84[dungeonMapRoomIter], mapId);
+        if (dungeonMapRoomIter + 1 < D_801F56B0.rooms) {
+            D_801F56B0.unk84[dungeonMapRoomIter + 1] =
+                ALIGN16((intptr_t)D_801F56B0.unk84[dungeonMapRoomIter] + MapData_CPID_GetSizeOfMapTex(mapId));
+        } else {
+            heapNext = (intptr_t)D_801F56B0.unk84[dungeonMapRoomIter] + MapData_CPID_GetSizeOfMapTex(mapId);
+        }
+    }
+    for (roomIter = 0; roomIter < D_801BEC1C; roomIter++) {
+        entry = &D_801BEBB8.minimapList->entry[roomIter];
+        if (entry->mapId == 0xFFFF) {
+            D_801F56B0.unk104[roomIter] = NULL;
+        } else {
+            var_s3 = false;
+            mapCompactId = MapData_GetMapCompactId(entry->mapId);
+            for (dungeonMapRoomIter = 0; dungeonMapRoomIter < D_801F56B0.rooms; dungeonMapRoomIter++) {
+                if (mapCompactId == D_801F56B0.unk4[dungeonMapRoomIter]) {
+                    var_s3 = true;
+                    break;
+                }
+            }
+            if (!var_s3) {
+                D_801F56B0.unk104[roomIter] = NULL;
+            } else {
+                void* dummy = D_801F56B0.unk84[dungeonMapRoomIter]; //! FAKE:
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_801064CC.s")
+                D_801F56B0.unk104[roomIter] = D_801F56B0.unk84[dungeonMapRoomIter];
+            }
+        }
+    }
+    return heapNext;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_map_disp/func_80106530.s")
+s32 func_80106BEC(s32 arg0, f32 arg1) {
```

Comment:
This function can return `true`/`false`, and every time it's called, can remove `!= 0` or `== 0` for `!`